st3/sublime_lib/resource_path.py

```diff
@@ -11,7 +10,35 @@
 __all__ = ['ResourcePath']
 
 
-@total_ordering
+ROOT_ORDER = {
+    'Packages': -2,
+    'Cache': -1,
+}
+
+
+def sort_by_load_order(paths):
```

Comment:
Would like to have a docstring here explaining roughly what it does (i.e. to what point it mirrors the internal order).

Case-insensitivity is missing, as far as I can tell.

Reply:
After more testing, it looks like the current implementation still sorts some paths differently from `sublime.find_resources` (e.g. `.DS_Store`). If `sublime.find_resources` is the closest we have to a canonical load order, then I may as well just modify `syntax` to use that ordering directly and remove all of the explicit sorting code. I wish I could be sure that that's absolutely correct, but it might be good enough to tide us over until the new API method obviates the whole thing.